### PR TITLE
[test] Remove `@required_version` decorator from tests

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

**Comment by vkarak** [comment]
@teojgo For this one, we need to make sure that the versions in the decorator are correct. I'm not sure that we maintained correctly these version numbers...

**Comment by teojgo** [comment]
> @teojgo For this one, we need to make sure that the versions in the decorator are correct. I'm not sure that we maintained correctly these version numbers...

Ok, I can look at the history to see when each test was introduced.

**Review by vkarak — CHANGES_REQUESTED** [review]
As discussed, I think it's better if we remove the annotations completely. We do not actually maintain them. They would make sense if we move the tests in a separate repo.

**Review by ekouts — APPROVED** [review]

File: cscs-checks/prgenv/mpi.py [hunk 1]
```diff
@@ -39,7 +38,7 @@ def __init__(self, required_thread):
         self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
                               'eiger:mc']
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu', 'PrgEnv-pgi',
-                                    'PrgEnv-intel', 'PrgEnv-aocc']
```
Comment (vkarak) [inline cscs-checks/prgenv/mpi.py:42]: Why did you remove `aocc`?

**Review by vkarak — APPROVED** [review]

File: cscs-checks/prgenv/mpi.py (hunk 1 above)
Comment (teojgo) [reply cscs-checks/prgenv/mpi.py:42]: There is an incompatibility with libsci

**Pull request merged**